syntax = "proto3";

package books;

// The book service definition.
service Books {
  // Get the list of books
  rpc ListBooks (ListBooksRequest) returns (ListBooksResponse) {}
}

// The request message
message ListBooksRequest {}

// The response message
message ListBooksResponse {
  repeated Book books = 1;
}

// The book structure
message Book {
  int32 book_id = 1;
  string title = 2;
  string author_name = 3;
}
